File: back-end/src/main/java/org/projectmanagement/model/dto/task/TaskCreateDto.java
package org.projectmanagement.model.dto.task;

import org.projectmanagement.model.enums.TaskPriority;
import org.projectmanagement.model.enums.TaskStatus;
import org.projectmanagement.model.enums.TaskType;

import java.sql.Timestamp;

public record TaskCreateDto(
    String title,
    String description,
    Integer backlog_item_id,
    Integer project_id,
    Integer sprint_id,
    TaskStatus status,
    TaskPriority priority,
    TaskType type,
    Integer story_points,
    String blockers,
    Double estimated_hours,
    Timestamp start_date,
    Timestamp end_date
) {
}

File: front-end/src/app/core/model/dto/task-create-dto.ts
 export interface TaskCreateDTO {
    title: string;
    description: string;
    project_id: number;
    sprint_id: number;
    backlog_item_id: number;
    priority: string;
    type: string;   
    estimated_hours: number;
    start_date: string;
    end_date: string;  
    status: string;
 }